File: drills/python/harnesses/nc104_word_dictionary.py
try:
    (WordDictionary)
except NameError:
    raise Exception("__signature_mismatch__")

__results__ = []
def __case__(label, expected, actual):
    __results__.append([label, repr(expected), repr(actual)])

__d__ = WordDictionary()
for __w__ in ["bad", "dad", "mad"]:
    __d__.addWord(__w__)

__case__("search('pad')", False, __d__.search("pad"))
__case__("search('bad')", True, __d__.search("bad"))
__case__("search('.ad')", True, __d__.search(".ad"))
__case__("search('b..')", True, __d__.search("b.."))
__case__("search('...')", True, __d__.search("..."))
__case__("search('b') -- too short", False, __d__.search("b"))
__case__("search('....') -- too long", False, __d__.search("...."))

File: drills/elixir/harnesses/nc104_word_dictionary.exs
store = Enum.reduce(["bad", "dad", "mad"], Solution.new(), &Solution.add_word(&2, &1))

[
  {~S{search("pad")}, inspect(false), inspect(Solution.search(store, "pad"))},
  {~S{search("bad")}, inspect(true), inspect(Solution.search(store, "bad"))},
  {~S{search(".ad")}, inspect(true), inspect(Solution.search(store, ".ad"))},
  {~S{search("b..")}, inspect(true), inspect(Solution.search(store, "b.."))},
  {~S{search("...")}, inspect(true), inspect(Solution.search(store, "..."))},
  {~S{search("b") -- too short}, inspect(false), inspect(Solution.search(store, "b"))},
  {~S{search("....") -- too long}, inspect(false), inspect(Solution.search(store, "...."))}
]
